src/SpecificDissipationRateEquationSystem.C

```diff
@@ -255,19 +257,19 @@ SpecificDissipationRateEquationSystem::register_interior_algorithm(
           nodeAlg.add_kernel<ScalarMassBDFNodeKernel>(realm_.bulk_data(), sdr_);
 
         if (SST == realm_.solutionOptions_->turbulenceModel_){
-          NaluEnv::self().naluOutputP0() << "call SDRSSTNodeKernel1: " <<std::endl;
           nodeAlg.add_kernel<SDRSSTNodeKernel>(realm_.meta_data());
-        }
-        else if ( (SST_DES == realm_.solutionOptions_->turbulenceModel_) || (SST_IDDES == realm_.solutionOptions_->turbulenceModel_ ) ){
+        } else if (
+          (SST_DES == realm_.solutionOptions_->turbulenceModel_) ||
+          (SST_IDDES == realm_.solutionOptions_->turbulenceModel_)) {
           nodeAlg.add_kernel<SDRSSTDESNodeKernel>(realm_.meta_data());
-        }
-        else if (SST_AMS == realm_.solutionOptions_->turbulenceModel_)
+        } else if (SST_AMS == realm_.solutionOptions_->turbulenceModel_) {
           nodeAlg.add_kernel<SDRSSTAMSNodeKernel>(
             realm_.meta_data(),
             realm_.solutionOptions_->get_coordinates_name());
-        else {
+        } else if (KO == realm_.solutionOptions_->turbulenceModel_) {
+          nodeAlg.add_kernel<SDRKONodeKernel>(realm_.meta_data());
+        } else {
           nodeAlg.add_kernel<SDRSSTNodeKernel>(realm_.meta_data());
```

Comment:
should we just throw an error then?

Reply:
Yeah throwing an error makes sense. We should never hit this condition since we are checking on an enum. It would be better if we used 'class enum' instead of enum to define these parameters in solution options though. That makes it so you can't pass an integer successfully which is how this could be broken.

Reply:
done.